test/cron/utils.test.ts

```diff
@@ -0,0 +1,43 @@
+import { assert } from 'chai';
+import config from 'config';
+import { has } from 'lodash';
+
+import { runCronJob } from '../../cron/utils';
+import { createRedisClient } from '../../server/lib/redis';
+
+describe('runCronJob', () => {
+  if (has(config, 'redis.serverUrl')) {
+    const clearRedis = async () => {
+      const redis = await createRedisClient();
+      await redis.del('lock:cron:test');
+    };
+    beforeEach(clearRedis);
+    afterEach(clearRedis);
+  }
+
+  it('should prevent two jobs from running at the same time', async () => {
+    let firstCompletes = false;
+    const first = runCronJob(
+      'test',
+      async () => {
+        await new Promise(resolve => setTimeout(resolve, 100));
+        firstCompletes = true;
+      },
+      10000,
+    );
+
+    let secondCompletes = false;
+    const second = runCronJob(
+      'test',
+      async () => {
+        await new Promise(resolve => setTimeout(resolve, 100));
+        secondCompletes = true;
+      },
+      10000,
+    );
```

Comment:
Running this locally gave me an error:
![image](https://github.com/user-attachments/assets/f8fdccd7-5eb9-42a1-97cf-c098817a6b52)

Looking at the code, there's not guarantee of execution order here: first and second are started in parallel, the same process tick. Suggestions:

```suggestion
    let firstCompletes = false;
    const first = runCronJob(
      'test',
      async () => {
        await new Promise(resolve => setTimeout(resolve, 100));
        firstCompletes = true;
      },
      10000,
    );

    await sleep(1); // Make sure the second job will start at the next process tick

    let secondCompletes = false;
    const second = runCronJob(
      'test',
      async () => {
        await new Promise(resolve => setTimeout(resolve, 100));
        secondCompletes = true;
      },
      10000,
    );
```